contrib_bots/lib/gcalendar.py

```diff
@@ -0,0 +1,254 @@
+
+    Timezones can be specified in both numerical (+00:00) or abbreviated
+    format (UTC).
+    If no timezone is specified, the server's will be considered as default.
+
+    Right now it only works for the calendar set as "primary" in your account
+    '''
+    def usage(self):
+        return '''
+            This plugin will allow users to create events
+            in a Google Calendar. Users should preface
+            messages with "@gcalendar"
+
+            Before running this, make sure to register a new
+            project in the Google Developers Console
+            (https://console.developers.google.com/start/api?id=calendar)
+            and download the "client secret" to ~/.gcal_client_secret.json
+            '''
+
+    def triage_message(self, message):
+        original_content = message['content']
+
+        is_gcalendar = original_content.startswith('@gcalendar')
+
+        return is_gcalendar
+
+    def handle_message(self, message, client, state_handler):
+        title, start_date, end_date = parse_message(message['content'])
+
+        if title is None or start_date is None: # There was a problem parsing
+            error = ('Unknown message format.\n'
+                     'Usage examples:\n\n'
+                     '```\n'
+                     '@gcalendar "Meeting with John" | Tuesday, 14 Jan 2003 13:37:00 | Tuesday, 14 Jan 2003 15:01:10\n'
+                     '@gcalendar Trip | 2003-01-14\n'
+                     '```\n')
+
+            reply_message(error, message, client)
+
+            return
+
+        event = {
+            'summary': title
+        }
+
+        if not end_date: # Full-day event
+            date = start_date.strftime('%Y-%m-%d')
+            event.update({
+                'start': {
+                    'date': date
+                },
+                'end': {
+                    'date': date
+                }
+            })
+        else:
+            event.update({
+                'start': {
+                    'dateTime': start_date.isoformat()
+                },
+                'end': {
+                    'dateTime': end_date.isoformat()
+                }
+            })
+
+        try:
+            # TODO: Choose calendar ID from somwhere
+            event = service.events().insert(calendarId='primary', body=event).execute()
+
+            date_format = '%c %Z'
+
+            start_str = start_date.strftime(date_format)
+            end_str = None
+
+            if not end_date:
+                reply = (':calendar: Full day event created!\n'
+                         '> **{title}**, on *{startDate}*')
+
+            else:
+                end_str = end_date.strftime(date_format)
+                reply = (':calendar: Event created!\n'
+                         '> **{title}**, from *{startDate}* to *{endDate}*')
+
+            reply_message(reply.format(title=title, startDate=start_str, endDate=end_str), message, client)
+        except errors.HttpError as e:
```

Comment:
What I mean is that if an `HttpError` has ocurred, I don't want the user to get the "event created" message (which is what I do in the code after `event = self.service.events().insert(calendarId='primary', body=event).execute()`.